### DISCUSSION TITLE
Value Converting to float

### DISCUSSION THREAD

**Discussion by binshadpix** [patrykandpatrick/vico#967] [Questions] [answered]
### Question

I have an issue related to amount values converting to float.
if the amount is greater than 7 digits and the entriesOf() function is converting the value to float.

when converting the values to float the value is being changed.
Is there any way to fix it in the 1.15 version?

it would be great if you give any alternate way to fix it in this version of 1.15.0

### Vico version(s)

1.15.0

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment, marked as answer]
Hello, @binshadpix. Based on a newer discussion you've opened, you've switched to Vico 2, but to address your question briefly:

Vico 1 uses `Float` internally. The behavior you're describing isn't an error but rather the nature of `Float`. Like any numerical data type, `Float` has limited precision. It's lower than that of e.g. `Double`. When a value that exceeds `Float`'s precision is converted to `Float`, some precision is lost. However, this is easy to address — when dealing with such values, you can first transform them for `Float` compatibility. This is possible in all realistic cases. It may involve subtracting the minimum value from all values, using a multiplier, and so on. The optimal approach depends on the nature of the data. These transformations can be undone as required, e.g. for formatting.

Vico 2 uses `Double` internally, providing greater precision. Thus, transformations generally aren't necessary anymore.
